File: aula1/server/src/main/java/br/edu/utfpr/pb/pw26s/server/service/CategoryService.java
package br.edu.utfpr.pb.pw26s.server.service;

import br.edu.utfpr.pb.pw26s.server.model.Category;

import java.util.List;

public interface CategoryService {

    List<Category> findAll();

    Category save(Category category);

    Category findOne(Long id);

    void delete(Long id);

}

File: aula2/client/src/services/CategoryService.js
import axios from 'axios';

const save = (category) => {
    return axios.post('/categories', category, {headers: getAuthHeader()});
}

const findAll = () => {
    return axios.get('/categories', {headers: getAuthHeader()});
}

const findOne = (id) => {
    return axios.get(`/categories/${id}`, {headers: getAuthHeader()});
}

const remove = (id) => {
    return axios.delete(`/categories/${id}`, {headers: getAuthHeader()});
}

const CategoryService = {
    save,
    findAll,
    findOne,
    remove
}

const getAuthHeader = () => {
    const token = JSON.parse(localStorage.getItem('token'));
    if (token) {
        return {Authorization: `Bearer ${token}`}; //'Bearer ' + token
    } else {
        return {}
    }
}

export default CategoryService;